src/omero_zarr/masks.py

```diff
@@ -45,201 +45,308 @@ def image_masks_to_zarr(image, args):
-        print(count)
-        for mask in shapes:
-            binim_yx, (t, c, z, y, x, h, w) = _mask_to_binim_yx(mask, dtype)
-            for i_t in _get_indices(ignored_dimensions, "T", t, size_t):
-                for i_c in _get_indices(ignored_dimensions, "C", c, size_c):
-                    for i_z in _get_indices(
-                        ignored_dimensions, "Z", z, size_z
+                labels.shape == mask_shape
+            ), "Invalid label shape: {}, expected {}".format(
+                labels.shape, mask_shape
+            )
+
+        for count, shapes in enumerate(masks):
+            # All shapes same color for each ROI
+            print(count)
+            for mask in shapes:
+                binim_yx, (t, c, z, y, x, h, w) = self._mask_to_binim_yx(mask)
+                for i_t in self._get_indices(
+                    ignored_dimensions, "T", t, size_t
+                ):
+                    for i_c in self._get_indices(
+                        ignored_dimensions, "C", c, size_c
                     ):
-                        if check_overlaps and np.any(
-                            np.logical_and(
-                                labels[
-                                    i_t, i_c, i_z, y : (y + h), x : (x + w)
-                                ].astype(np.bool),
-                                binim_yx,
-                            )
+                        for i_z in self._get_indices(
+                            ignored_dimensions, "Z", z, size_z
                         ):
-                            raise Exception(
-                                "Mask {} overlaps with existing labels".format(
-                                    count
+                            if check_overlaps and np.any(
+                                np.logical_and(
+                                    labels[
+                                        i_t, i_c, i_z, y : (y + h), x : (x + w)
+                                    ].astype(np.bool),
+                                    binim_yx,
                                 )
+                            ):
+                                raise Exception(
+                                    (
+                                        f"Mask {count} overlaps "
+                                        "with existing labels"
+                                    )
+                                )
+                            # ADD to the array, so zeros in our binarray don't
+                            # wipe out previous masks
+                            labels[
+                                i_t, i_c, i_z, y : (y + h), x : (x + w)
+                            ] += (
+                                binim_yx * (count + 1)  # Prevent zeroing
+                            )
+
+        return labels
+
+    def stack_masks(
+        self,
+        masks,
+        mask_shape,
+        ignored_dimensions=None,
+        check_overlaps=True,
+        target=None,
```

Comment:
This is a required parameter:
```
        if not target:
            raise Exception("No target")
```
Is there a reason it defaults to `None` instead of having no default and therefore being required?

Reply:
Just due to the ordering. I was trying to keep the original diff simple before I decided to support all 3 methods, and you can't have a non-optional after other options. I'll just move the argument up.